Sources/FoundationEssentials/Subprocess/Platforms/Subprocess+Unix.swift

```diff
@@ -0,0 +1,286 @@
+    }
+
+    public func sendSignal(_ signal: Signal, toProcessGroup shouldSendToProcessGroup: Bool) throws {
+        let pid = shouldSendToProcessGroup ? -(self.processIdentifier.value) : self.processIdentifier.value
+        guard kill(pid, signal.rawValue) == 0 else {
+            throw POSIXError(.init(rawValue: errno)!)
+        }
+    }
+}
+
+// MARK: Environment Resolution
+extension Subprocess.Environment {
+    internal static let pathEnvironmentVariableName = "PATH"
+
+    internal func pathValue() -> String? {
+        switch self.config {
+        case .inherit(let overrides):
+            // If PATH value exists in overrides, use it
+            if let value = overrides[.string(Self.pathEnvironmentVariableName)] {
+                return value.stringValue
+            }
+            // Fall back to current process
+            return ProcessInfo.processInfo.environment[Self.pathEnvironmentVariableName]
+        case .custom(let fullEnvironment):
+            if let value = fullEnvironment[.string(Self.pathEnvironmentVariableName)] {
+                return value.stringValue
+            }
+            return nil
+        }
+    }
+
+    // This method follows the standard "create" rule: `env` needs to be
+    // manually deallocated
+    internal func createEnv() -> [UnsafeMutablePointer<CChar>?] {
+        func createFullCString(
+            fromKey keyContainer: Subprocess.StringOrRawBytes,
+            value valueContainer: Subprocess.StringOrRawBytes
+        ) -> UnsafeMutablePointer<CChar> {
+            let rawByteKey: UnsafeMutablePointer<CChar> = keyContainer.createRawBytes()
+            let rawByteValue: UnsafeMutablePointer<CChar> = valueContainer.createRawBytes()
+            defer {
+                rawByteKey.deallocate()
+                rawByteValue.deallocate()
+            }
+            /// length = `key` + `=` + `value` + `\null`
+            let totalLength = keyContainer.count + 1 + valueContainer.count + 1
+            let fullString: UnsafeMutablePointer<CChar> = .allocate(capacity: totalLength)
+            #if canImport(Darwin)
+            _ = snprintf(ptr: fullString, totalLength, "%s=%s", rawByteKey, rawByteValue)
+            #else
+            _ = _shims_snprintf(fullString, CInt(totalLength), "%s=%s", rawByteKey, rawByteValue)
+            #endif
+            return fullString
+        }
+
+        var env: [UnsafeMutablePointer<CChar>?] = []
+        switch self.config {
+        case .inherit(let updates):
+            var current = ProcessInfo.processInfo.environment
+            for (keyContainer, valueContainer) in updates {
+                if let stringKey = keyContainer.stringValue {
+                    // Remove the value from current to override it
+                    current.removeValue(forKey: stringKey)
+                }
+                // Fast path
+                if case .string(let stringKey) = keyContainer,
+                   case .string(let stringValue) = valueContainer {
```

Comment:
Previously, `Task` used the `fileSystemRepresentation` of all environment keys/values when spawning a process. Do we want to maintain that behavior with `Subprocess` or do we intentionally want to avoid doing that here?